### PR TITLE
feat: Added Side Pagination button #2071

### LINKED ISSUE #2071: [Feature] Add pagination buttons at the top-right corner
### Description

Adding pagination buttons at the top-right corner would lead to a great user experience, as the user won't have to scroll down to bottom to navigate through different pages.

Add the buttons in top-right corner below the social media icons, and do NOT remove the pagination buttons from the bottom.

### Screenshots

IMG_20231022_180024

### Checklist

- [X] I have checked the existing issues
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [ ] I am willing to work on this issue (optional)
- [ ] I am participating in Hacktoberfest 2023

### PR DESCRIPTION
**Pull request by Raj100679**
## Fixes Issue

Closes #2071 

## Changes proposed
Added Pagination button on the Side

## Screenshots
Screenshot (113)

### CODE REVIEW

**Review by rupali-codes — COMMENTED** [review]
Hey @Raj100679, can we make it horizontal instead, just right above the cards, in top-right corner

**Comment by Raj100679** [comment]
So should I make it horizontal when screen size gets small or for every screen size

**Comment by Raj100679** [comment]
@rupali-codes @Anmol-Baranwal 
topSide

**Comment by Anmol-Baranwal** [comment]
> @rupali-codes @Anmol-Baranwal 

@rupali-codes What do you think?

This somehow feels weird due to next and prev as far as I believe.

**Comment by rupali-codes** [comment]
@Raj100679 & @Anmol-Baranwal can we make it smaller? and reduce the gap, then it would look better I guess

**Comment by Raj100679** [comment]
@rupali-codes @Anmol-Baranwal 
Screenshot (117)
Is this weird?

**Comment by Raj100679** [comment]
@rupali-codes @Anmol-Baranwal can you check it?

**Comment by CBID2** [comment]
> @rupali-codes @Anmol-Baranwal can you check it?

It looks ok to me! :)

**Comment by rupali-codes** [comment]
Looks fine to me.

**Review by rupali-codes — DISMISSED** [review]

**Comment by rupali-codes** [comment]
@Anmol-Baranwal and @CBID2 are you up?

**Review by CBID2 — APPROVED** [review]
LGTM

**Comment by Raj100679** [comment]
@rupali-codes I guess , I haven't pushed this changes to the PR will update the PR for merging purpose.

**Comment by Anmol-Baranwal** [comment]
> @rupali-codes I guess , I haven't pushed this changes to the PR will update the PR for merging purpose.

Please do it. It is easier for review purpose.

**Comment by Raj100679** [comment]
@rupali-codes @Anmol-Baranwal I have pushed the changes , Thanks

**Review by Anmol-Baranwal — COMMENTED** [review]
@rupali-codes 
What do you think?

image

This may not be appropriate, it should be only at one place right.

**Review by rupali-codes — COMMENTED** [review]
@Anmol-Baranwal It looks fine to me,

**Comment by aftabrehan** [comment]
> @rupali-codes What do you think?
> 
> image
> 
> This may not be appropriate, it should be only at one place right.

@Anmol-Baranwal, Why are there two Paginations?

**Comment by Anmol-Baranwal** [comment]
> > @rupali-codes What do you think?
> > image
> > This may not be appropriate, it should be only at one place right.
> 
> @Anmol-Baranwal, Why are there two Paginations?

There isn't supposed to be, but no worries.

We can handle it quickly. The PR has been open for too long, and we shouldn't make the author wait any longer.

**Comment by aftabrehan** [comment]
> > > @rupali-codes What do you think?
> > > image
> > > This may not be appropriate, it should be only at one place right.
> > 
> > @Anmol-Baranwal, Why are there two Paginations?
> 
> There isn't supposed to be, but no worries.
> 
> We can handle it quickly. The PR has been open for too long, and we shouldn't make the author wait any longer.

Okay, make sense.

File: components/Pagination/Pagination.tsx [hunk 1]
```diff
@@ -1,43 +1,59 @@
...
-          <div className="flex items-center px-6 py-1 gap-4">
+          <div className={
+            toporbottom == true
+              ? clsx(
+                "flex items-center px-6 py-1 gap-4"
+                )
+              : clsx(
+                "flex items-center gap-2"
+                )
+          }>
```
Comment (aftabrehan) [inline components/Pagination/Pagination.tsx:56]: `clsx` is supposed to render multiple classes. We don't need to render it multiple times.

```suggestion
          <div className={clsx(
            'flex items-center',
            toporbottom == true ? 'px-6 py-1 gap-4' : 'gap-2'
           )}>
```

File: pages/[category]/[...subcategory].tsx [hunk 2]
```diff
@@ -98,11 +99,18 @@ const SubCategory: NextPage<PageProps> = ({ subcategory }) => {
       <TopBar className="shadow-black-500/50 fixed top-[76px] z-30 flex w-full -translate-x-4 items-center bg-gray-100 px-4 pt-6 pb-4 shadow-xl dark:bg-gray-900 md:hidden" />
       <div
         data-custom="restrict-click-outside"
-        className="relative min-h-[calc(100%-68px)] w-full pt-[85px] pb-4 md:min-h-[calc(100%-76px)] md:px-10 md:pt-10"
+        className=" relative min-h-[calc(100%-68px)]  w-full pt-[85px] pb-4 md:min-h-[calc(100%-76px)] md:px-10 md:pt-10"
```
Comment (aftabrehan) [inline pages/[category]/[...subcategory].tsx:102]: Please, avoid adding extra spaces.

```suggestion
        className="relative min-h-[calc(100%-68px)] w-full pt-[85px] pb-4 md:min-h-[calc(100%-76px)] md:px-10 md:pt-10"
```

**Review by aftabrehan — COMMENTED** [review]
Excellent PR, @Raj100679 💪 🚀 💯 

The PR is okay, it just needs minor adjustments and then we're good to go 🚀

File: pages/[category]/[...subcategory].tsx [hunk 3]
```diff
@@ -98,11 +99,18 @@ const SubCategory: NextPage<PageProps> = ({ subcategory }) => {
...
       >
         {content}
-        <div className="min-w-full h-10 py-5"/>
         <Pagination
+          toporbottom={toporbottom}
+          totalPages={totalPages}
+          currentPage={currentPage}
+          handlePageChange={handlePageChange}
+        />
+        <div className=" min-w-full h-10 py-5" />
```
[the remark this answers is not in the document]
Comment (Raj100679) [reply pages/[category]/[...subcategory].tsx:111]: @aftabrehan , can you suggest what changes do u need? 
I can still work on it, I just forgot about it

[the remark this answers is not in the document]
Comment (aftabrehan) [reply pages/[category]/[...subcategory].tsx:111]: Is this empty div still required? I mean there are still two paginations, we can remove one of them, right?

[the remark this answers is not in the document]
Comment (CBID2) [reply pages/[category]/[...subcategory].tsx:111]: Are you still working on this @Raj100679?

File: pages/[category]/[...subcategory].tsx [hunk 4]
```diff
@@ -33,6 +33,7 @@ const SubCategory: NextPage<PageProps> = ({ subcategory }) => {
   } else {
     content = <ComingSoon />
   }
+  const toporbottom = true
```
[the remark this answers is not in the document]
Comment (CBID2) [reply pages/[category]/[...subcategory].tsx:36]: What do you think @Raj100679?

File: pages/[category]/[...subcategory].tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Raj100679) [reply pages/[category]/[...subcategory].tsx:111]: Hey, I will work on this

File: components/Pagination/Pagination.tsx [hunk 5]
```diff
@@ -1,43 +1,54 @@
...
+            toporbottom == true
+              ? clsx(
+                  'w-full z-20 flex lg:w-full items-center justify-center ',
+                  'absolute bottom-2 right-0'
+                )
+              : clsx(
+                  ' z-20 flex w-full lg:w-full items-center justify-end ',
+                  'absolute top-0 right-0 '
+                )
+          }
```
Comment (aftabrehan) [inline components/Pagination/Pagination.tsx:46]: I think we can achieve this in a single `clsx` wrapper.

```suggestion
          className={clsx(
            toporbottom == true
              ?  'w-full z-20 flex lg:w-full items-center justify-center absolute bottom-2 right-0'
              : 'z-20 flex w-full lg:w-full items-center justify-end absolute top-0 right-0'
           )}
```

**Review by aftabrehan — CHANGES_REQUESTED** [review]

**Comment by CBID2** [comment]
@Raj100679, any updates?

**Comment by Raj100679** [comment]
@CBID2 I will work on it today surely, Sorry for not replying

**Comment by Anmol-Baranwal** [comment]
> @CBID2 I will work on it today surely, Sorry for not replying

The PR has been stale for very long. Please try to finish it as soon as possible.

**Comment by Raj100679** [comment]
@CBID2 Created a new pr, Can you check it, closing it here

**Pull request closed without merging**